sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/export/__init__.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import json
+import logging
+import typing
+from enum import Enum
+from urllib.parse import urlparse
+
+from opentelemetry.sdk.metrics.export import MetricsExportResult
+from opentelemetry.sdk.trace.export import SpanExportResult
+from opentelemetry.sdk.util import ns_to_iso_str
+from opentelemetry.trace import Span, SpanKind
+from opentelemetry.trace.status import StatusCanonicalCode
+
+from azure.core.exceptions import HttpResponseError
+from azure.core.pipeline.policies import ProxyPolicy, RetryPolicy
+from opentelemetry.exporter.azuremonitor import utils
+from opentelemetry.exporter.azuremonitor._generated import AzureMonitorExporter
+from opentelemetry.exporter.azuremonitor._generated.models import TelemetryItem, TrackResponse
+from opentelemetry.exporter.azuremonitor.options import ExporterOptions
+from opentelemetry.exporter.azuremonitor.storage import LocalFileStorage
+
+logger = logging.getLogger(__name__)
+
+
+class ExportResult(Enum):
+    SUCCESS = 0
+    FAILED_RETRYABLE = 1
+    FAILED_NOT_RETRYABLE = 2
+
+
+# pylint: disable=broad-except
+class BaseExporter:
+    """Azure Monitor base exporter for OpenTelemetry.
+
+    Args:
+        options: :doc:`export.options` to allow configuration for the exporter
+    """
+
+    def __init__(self, **options):
+        self._telemetry_processors = []
+        self.options = ExporterOptions(**options)
+        retry_policy = RetryPolicy(timeout=self.options.timeout)
```

Comment:
users should also have the option to provide their own retry policy.
perhaps something like 
```suggestion
        retry_policy = kwargs.get('retry_policy', RetryPolicy(timeout=self.options.timeout))
```

Reply:
I'm not sure if we want to provide the users with the option to provide their own `Policy`. This seems like an Azure specific concept and requires knowledge of what `Policies` exist and what their api surface is like. I think instead we can a set list of configurable "options" (like in `ExporterOptions`) that relate to things they can change within the `Policies` that the exporter uses. The benefit of this is we don't expose a `**kwargs` in the constructor (which seems "hacky", we should be explicit in the configuration options we provide users). The downside is we might bloat up `ExporterOptions`. Thoughts?

Reply:
I would incline towards having kwargs where users have the option to send in their own policies only because of the language precendent set in other python libraries in this repo.
Also, bloating Exporter options would prolly be a concern. One of the guidelines is to not have signatures with more than 5 positional params and we need to use kwargs if it does
https://azure.github.io/azure-sdk/python_introduction.html#method-signatures